util.grace

```diff
@@ -396,31 +396,38 @@ method execDir {
     execDirCache
 }
 
-method file(name) on(origin) orPath(pathString) otherwise(action) {
+method files(names) on(origin) orPath(pathString) otherwise(action) {
     def locations = filePath.split(pathString)
     locations.addFirst(origin)
     if (locations.contains(outDir).not) then { locations.addFirst(outDir) }
         // it's important that outDir is first; if not, the compiler
         // might compile an imported file, but then be unable to find the
         // code that it just generated.
     if (locations.contains(execDir).not) then { locations.addLast(execDir) }
-    def candidate = name.copy
-    def originalDir = name.directory
-    if ((originalDir.size > 0) && {originalDir.first == "/"}) then {
-        if (candidate.exists) then {
-            return candidate
-        } else {
-            return action.apply ""
+    def candidates = sequence << names.map{n -> n.copy}
+    def originalDirs = sequence << names.map{n -> n.directory}
+    // This assumes that either everything in "names" is an absolute path, or nothing is
+    if ((names.size > 0) && {originalDirs.at(1).size > 0} && {originalDirs.at(1).first == "/"}) then {
+        names.indices.do { i ->
+            if (candidates.at(i).exists) then {
+               return candidates.at(i)
+            }
         }
+        return action.apply ""
     }
     locations.do { each ->
-        candidate.setDirectory(each ++ originalDir)
-        if ( candidate.exists ) then {
-            return candidate
+        names.indices.do { i ->
+            def candidate = candidates.at(i).copy.setDirectory(each ++ originalDirs.at(i))
+            if ( candidate.exists ) then {
+               return candidate
```

Comment:
I was expecting to see a sequence of candidates returned, one for each name in `names`